roles/docker/tasks/main.yml

```diff
@@ -22,7 +22,7 @@
 
 - name: Install docker
   action: "{{ ansible_pkg_mgr }} name=docker{{ '-' + docker_version if docker_version is defined and docker_version != '' else '' }} state=present"
-  when: not openshift.common.is_atomic | bool and not docker_version_result | skipped and docker_version_result.stdout | default('0.0', True) | version_compare(docker_version, 'lt')
+  when: not openshift.common.is_atomic | bool and docker_downgrade_result | skipped
```

Comment:
As far as I can tell this covers the previous case where we only want to do the install if the version available is less than the version we want and it fixes the issue where docker wasn't being installed when docker_version is not present (since docker_downgrade_result will only be skipped if docker_version_result is skipped or the installed version is not greater than the current version). Now it will not skip this task when the version installed is equal to the version wanted, but that is a noop.